### PR TITLE
fix(launcher-api): remove orphan JobResults visibility permissions

### PR DESCRIPTION
**Pull request by mabw-rte**
Context: 
In the job listing of `AntaREST`, we call two endpoints: 
- `GET`, `v1/launcher/jobs`
- `GET`, `v1/launcher/jobs/{job_id}/progress`
It turns out that when user does not have a permission for some job, this job, though listed (using the first endpoint of `jobs`), spits out an error `403` when it is called for the second endpoint `progress`.

Issue:
For the endpoint `jobs`, the logic beyond the permissions includes a second option where when an orphan job is created recently enough would become visible to everyone. This logic was not included in the `progress` endpoint which leads to the `Exception` above. 

Solution:
After consulting the application users, they think that this `orphan` visibility logic is not that much relevant and should be removed (at least temporarily).

### CODE REVIEW

File: antarest/launcher/service.py [hunk 1]
```diff
@@ -305,7 +305,6 @@ def _filter_from_user_permission(self, job_results: List[JobResult], user: Optio
         if not user:
             return []
 
-        orphan_visibility_threshold = datetime.utcnow() - timedelta(days=ORPHAN_JOBS_VISIBILITY_THRESHOLD)
```
Comment (MartinBelthle) [inline antarest/launcher/service.py:308]: You can also remove the constant as we don't use it anymore

**Review by MartinBelthle — CHANGES_REQUESTED** [review]
Seems good to me just a minor change

**Comment by MartinBelthle** [comment]
You can also remove the import of `timedelta` that we don't use anymore I believe

**Review by sylvlecl — APPROVED** [review]

**Review by MartinBelthle — APPROVED** [review]

**Review by mabw-rte — COMMENTED** [review]
@sylvlecl, il faudrait priviligier de `rebase` en local ... Tu pourrais voir avec @skamril, on a deja eu la discussion

**Pull request merged**